<launch>

	<arg name="number_of_worlds" default="4"/>
	<!-- args for changing terrain params -->
	<arg name="cage_height_range_begin" default="0.02"/>
	<arg name="cage_height_range_end" default="0.165" />
	<arg name="cage_width_and_lengh" default="0.2"/> <!-- need to be float type -->
	<arg name="cell_width_number" default="18"/>
	<arg name="cell_length_number" default="18" />
	<arg name="cage_height_param" default="rand"/>
	<arg name="terrain_file_path_without_file_name" default="$(find strirus_ga_body_optimization)/maps/Generated_terrain"/>

	<!-- args for changing world params -->
	<arg name="world_file_path_without_extention" default="$(find strirus_ga_body_optimization)/worlds/testing_area"/>
	<arg name="physics_iter" default="900" />
	<arg name="sor" default="0.9" />
	<arg name="real_time_update_rate" default="1000"/>
	<arg name="max_step_size" default="0.003"/>
	<arg name="package_name" default="strirus_ga_body_optimization"/>
	<arg name="terrain_path" default="maps/Generated_terrain" /> <!-- without first '/' -->


	<node name="whole_worlds_generation" pkg="strirus_ga_body_optimization" type="whole_worlds_generation.py" output="screen" required="false">
		<param name="number_of_worlds" value="$(arg number_of_worlds)"/>
		
		<param name="cage_height_range_begin" value="$(arg cage_height_range_begin)"/>
		<param name="cage_height_range_end" value="$(arg cage_height_range_end)" />
		<param name="cage_width_and_lengh" value="$(arg cage_width_and_lengh)"/> 
		<param name="cell_width_number" value="$(arg cell_width_number)"/>
		<param name="cell_length_number" value="$(arg cell_length_number)" />
		<param name="cage_height_param" value="$(arg cage_height_param)"/>
		<param name="terrain_file_path_without_file_name" value="$(arg terrain_file_path_without_file_name)"/>

		<param name="world_file_path_without_extention" value="$(arg world_file_path_without_extention)"/>
		<param name="physics_iter" value="$(arg physics_iter)" />
		<param name="sor" value="$(arg sor)" />
		<param name="real_time_update_rate" value="$(arg real_time_update_rate)"/> 
		<param name="max_step_size" value="$(arg max_step_size)"/>
		<param name="package_name" value="$(arg package_name)"/>
		<param name="terrain_path" value="$(arg terrain_path)" />
	</node>

</launch>
